### PR TITLE
hack/e2e.go: listen for Interrupt signal and attempt cleanup if run with --down

### PR DESCRIPTION
**Pull request by jessfraz**
**What this PR does / why we need it**: adds a listener on ^C, so if `hack/e2e.go` is run with `--down` and the user ^C, then it will attempt to cleanup the resources.

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #

**Special notes for your reviewer**:

**Release note**:

``` release-note
NONE
```

Signed-off-by: Jess Frazelle me@jessfraz.com

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32154)

### CODE REVIEW

**Comment by jessfraz** [comment]
calm yourself bot I'll rebase

**Comment by jessfraz** [comment]
@k8s-bot kubemark e2e test this

File: hack/e2e.go [hunk 1]
```diff
@@ -150,6 +151,20 @@ func main() {
...
 	flag.Parse()
 
+	if *down {
+		// listen for signals such as ^C and gracefully attempt to clean up
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		go func() {
+			for range c {
+				log.Print("Captured ^C, gracefully attempting to cleanup resources..")
+				if !TearDown() {
```
Comment (eparis) [inline hack/e2e.go]: Maybe I'm just too dumb to review, but I can't find `func TearDown` anywhere...

**Comment by jessfraz** [comment]
I think this code was refactored I'll find whatever it's named now

**Comment by eparis** [comment]
My guess is a little re-organization and use `deploy.Down()`    But what do I know  :)

**Comment by jessfraz** [comment]
you are correct :)

**Comment by jessfraz** [comment]
@k8s-bot kubemark e2e test this

**Pull request merged**